tests/autofill/autofill_plugin.test.ts

```diff
@@ -218,14 +218,119 @@ describe("Autofill", () => {
-      expect(getCellText(model, "A4")).toBe("3/6/2003");
-      expect(getCellText(model, "A5")).toBe("3/7/2003");
-      expect(getCellText(model, "A6")).toBe("3/8/2003");
+    describe("Autofill dates", () => {
+      test("consecutive dates", () => {
+        setCellContent(model, "A1", "3/28/2003");
+        setCellContent(model, "A2", "3/29/2003");
+        setCellContent(model, "A3", "3/30/2003");
+        autofill("A1:A3", "A6");
+        expect(getCellText(model, "A4")).toBe("3/31/2003");
+        expect(getCellText(model, "A5")).toBe("4/1/2003");
+        expect(getCellText(model, "A6")).toBe("4/2/2003");
+      });
+
+      test("Descending dates", () => {
+        setCellContent(model, "A1", "3/4/2003");
+        setCellContent(model, "A2", "3/3/2003");
+        autofill("A1:A2", "A5");
+        expect(getCellText(model, "A3")).toBe("3/2/2003");
+        expect(getCellText(model, "A4")).toBe("3/1/2003");
+        expect(getCellText(model, "A5")).toBe("2/28/2003");
+      });
+
+      test("Autofill upwards consecutive dates", () => {
+        setCellContent(model, "A4", "3/31/2003");
+        setCellContent(model, "A5", "4/1/2003");
+        setCellContent(model, "A6", "4/2/2003");
+        autofill("A4:A6", "A1");
+        expect(getCellText(model, "A1")).toBe("3/28/2003");
+        expect(getCellText(model, "A2")).toBe("3/29/2003");
+        expect(getCellText(model, "A3")).toBe("3/30/2003");
+      });
+
+      test("dates with consistent day gap", () => {
+        setCellContent(model, "A1", "4/21/2003");
+        setCellContent(model, "A2", "4/23/2003");
+        setCellContent(model, "A3", "4/25/2003");
+        autofill("A1:A3", "A7");
+        expect(getCellText(model, "A4")).toBe("4/27/2003");
+        expect(getCellText(model, "A5")).toBe("4/29/2003");
+        expect(getCellText(model, "A6")).toBe("5/1/2003");
+        expect(getCellText(model, "A7")).toBe("5/3/2003");
+      });
+
+      test("dates with consistent month gap", () => {
+        setCellContent(model, "A1", "3/24/2003");
+        setCellContent(model, "A2", "5/24/2003");
+        setCellContent(model, "A3", "7/24/2003");
+        autofill("A1:A3", "A6");
+        expect(getCellText(model, "A4")).toBe("9/24/2003");
+        expect(getCellText(model, "A5")).toBe("11/24/2003");
+        expect(getCellText(model, "A6")).toBe("1/24/2004");
+      });
+
+      test("dates with consistent year gap", () => {
+        setCellContent(model, "A1", "3/24/2000");
+        setCellContent(model, "A2", "3/24/2003");
+        setCellContent(model, "A3", "3/24/2006");
+        autofill("A1:A3", "A6");
+        expect(getCellText(model, "A4")).toBe("3/24/2009");
+        expect(getCellText(model, "A5")).toBe("3/24/2012");
+        expect(getCellText(model, "A6")).toBe("3/24/2015");
+      });
+
+      test("dates 2 year apart with leap year", () => {
+        setCellContent(model, "A1", "3/24/2000");
+        setCellContent(model, "A2", "3/24/2002");
```

Comment:
You should pick a date before march to test the leap year no ? There's no 29 february between the two dates you picked :)

Reply:
doesn't change anything, i'm bound to "pass though" the 29 of february either between 2002 and 2004 or 2004 and 2006